boot/bootutil/src/loader.c

```diff
@@ -1903,6 +1905,60 @@ boot_update_hw_rollback_protection(struct boot_loader_state *state)
 #endif
 }
 
+/**
+ * Checks test swap downgrade prevention conditions.
+ *
+ * Function called only for swap upgrades test run.  It may prevent
+ * swap if slot 1 image has <= version number or < security counter
+ *
+ * @param  state        Boot loader status information.
+ *
+ * @return              0 - image can be swapped, -1 downgrade prevention
+ */
+static int
+check_downgrade_prevention(struct boot_loader_state *state)
+{
+#if defined(MCUBOOT_DOWNGRADE_PREVENTION) && \
+    (defined(MCUBOOT_SWAP_USING_MOVE) || defined(MCUBOOT_SWAP_USING_SCRATCH))
```

Comment:
In an upgrade only system e.g. nrf53 with netcore update support, assuming code is stored in external QSPI, doesn't that mean that if a person reloads an older firmware by flashing it to the QSPI directly then there is no way for mcuboot to block the overwrite? Or perhaps using serial recovery?

Reply:
@nordicjm If someone has direct access to the HW at such level we can't do anything about it: this is out of mcuboot itself focus.
Serial recovery objective is giving the user total control on image(es).